components/XOGame.js

```diff
@@ -17,22 +28,54 @@ export default function XOGame() {
     ["", "", ""],
   ]);
 
+  const [isCPUNext, setIsCPUNext] = useState(false);
+
   function playFieldLocation(arrayIndex, index) {
     
     
     board[arrayIndex][index] = players?.HUMAN?.SYM;
     setBoard((board) => [...board]);
     
-    
+    setIsCPUNext(true);
+  }
+
+  useEffect(() => {
+    if (isCPUNext) {
+      cPUPlay();
+    }
+  }, [isCPUNext]);
+
+  function cPUPlay() {
+    sleep(1000);
```

Comment:
It's generally not advisable to block the JS execution.
I would rather use a `setTimeout` function within the `useEffect()`, something like:
```
useEffect(() => {
    if (isCPUNext) {
      setTimeout(cPUPlay, 1000);
    }
  }, [isCPUNext]);
  ```
Then you could remove the `sleep(1000)` part in your `cpuPlay()` function.

As a consequence you might want to check in the player´s turn function that it´s not the CPU´s turn when the player attempts to click a second time within the time window.
```
  function playFieldLocation(arrayIndex, index) {
    if (!isCPUNext) {
```

Reply:
Thanks for the input